cxx-sensors/src/test/java/org/sonar/cxx/sensors/coverage/CxxTestwellCtcTxtParserTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Sonar C++ Plugin (Community)
+ * Copyright (C) 2010-2018 SonarOpenCommunity
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.cxx.sensors.coverage;
+
+import org.assertj.core.api.SoftAssertions;
+import org.junit.Before;
+import org.junit.Test;
+import static org.mockito.Mockito.when;
+
+import org.sonar.api.batch.fs.internal.DefaultFileSystem;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.config.internal.MapSettings;
+import org.sonar.cxx.CxxLanguage;
+import org.sonar.cxx.sensors.utils.TestUtils;
+
+public class CxxTestwellCtcTxtParserTest {
+
+  private CxxCoverageSensor sensor;
+  private DefaultFileSystem fs;
+  private SensorContextTester context;
+  private CxxLanguage language;
+  private final MapSettings settings = new MapSettings();
+
+  @Before
+  public void setUp() {
```

Comment:
It is very error-prone to share mutable state between test cases
maybe it's convenient (?) to declare ...

+  private CxxCoverageSensor sensor;
+  private DefaultFileSystem fs;
+  private SensorContextTester context;

... and permanently reset them in `@Before`...

but sharing of...

+ private final MapSettings settings = new MapSettings();

... is just wrong IMHO

```
private static MapSettings singletonMapSettings( String key, String value )
{
     MapSettings settings = new MapSettings();
     settings.set( key, value );
     return settings;
}

@Test
public void myWhateverTest() {
  MapSettings settings = singletonMapSettings( "some key", "some value );
}
```

P. S.  I grepped in other tests for `settings` and these pattern seems to be common (which still doesn't make it correct). Feel free to ignore my comment.